### PR TITLE
Track benchmarks on `main` branch using Bencher

### PR DESCRIPTION
**Pull request by epompeii**
These changes add tracking of the metrics for the `main` branch of `rustls` using Bencher.

In order to accomplish this, three configuration keys were added:
- `bencher_api_token`: An API token for https://bencher.dev
- `bencher_project_id`: (Optional) It sets the Bencher Project UUID or slug and defaults to `rustls`
- `bencher_testbed_id`: (Optional) It sets the Bencher Testbed UUID or slug and defaults to `benchmarking-host` (I'm very much up for suggestions on what the default should be here)

The `BencherClient` is constructed and stored in the `EventQueue` using the `bencher_api_token` key.
While the other two are used (if provided) at runtime to create and send the report to Bencher.

There were some minor formatting changes and whitespace cleanup done by my IDE automatically. If you would like for me to go back and revert these, please let me know.

### CODE REVIEW

**Comment by aochagavia** [comment]
Thanks for this! Do you mind if take the changes and adapt them myself to what we need (making sure you are still mentioned as one of the authors of the changes)? That would save us some back-and-forth.

**Comment by epompeii** [comment]
Sounds good! Feel free to edit this PR or whatever workflow works best for you.

File: ci-bench-runner/Cargo.toml [hunk 1]
```diff
@@ -6,17 +6,28 @@ edition = "2021"
 [dependencies]
 anyhow = "1.0.75"
 axum = "0.6.20"
+bencher_client = { git = "https://github.com/bencherdev/bencher", rev = "d0532976" }
```
Comment (aochagavia) [inline ci-bench-runner/Cargo.toml:9]: Are there any plans for a crates.io release? Depending on a GitHub repository is less friendly to reproducible builds

Comment (epompeii) [reply ci-bench-runner/Cargo.toml:9]: Definitely! That has been on my todo (https://github.com/bencherdev/bencher/issues/66). If it is a blocker here or even just a nice to have, I can prioritize getting it done.

Pinning things to a `rev` should keep them reproducible. Though, I definitely understand the preference for a crates.io release.

Comment (aochagavia) [reply ci-bench-runner/Cargo.toml:9]: I agree that a `rev` helps, and I guess the repo won't disappear anytime soon, but it's good to know a crates.io release is coming at some point

**Comment by aochagavia** [comment]
Thanks again for this! I ended up cherry-picking bits of this PR into a new branch, which resulted in https://github.com/aochagavia/rustls-bench-app/pull/6

**Pull request closed without merging**